gobblin-test-harness/src/main/java/gobblin/test/execution/validator/Validator.java

```diff
@@ -0,0 +1,28 @@
+package gobblin.test.execution.validator;
+
+import java.util.Comparator;
+
+/**
+ *  An interface is for defining the validator for the test. The input can be the set of input files and the output will be be output of 
+ * 
+ * @author sveerama
+ *
+ * @param <I>
+ * @param <O>
+ */
+
+public interface Validator<I, O> extends Comparator<O>
```

Comment:
I'm not sure what the output of the validator would be. I would guess that the output of the validator would be something like a boolean indicating if the validation succeeded or failed, or it could be a list of ValidationErrors that were hit during the validation phase.

Also, it may make more sense to have an interface called `ValidationRule` where each implementation of the interface is a different validation step. Then the `Validator` is just responsible for executing the `ValidationRule`s.

Also, does this need to implement `Comparator`. The `Comparator` interface is typically used for use cases such as sorting.